### DISCUSSION TITLE
Use of CombinedRelationManagerTabsWithContent

### DISCUSSION THREAD

**Discussion by MagicDidou** [filamentphp/filament#12237] [Help] [answered]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

Hi everybody,

I'm facing an issue I cannot solve and can't find any solution on the web. I have a Resource (Job) with a relation (Hours) which is managed through a RelationMabager.

I'm using the `hasCombinedRelationManagerWithContent() { return true ; }` override to have the relation manager and the edit form on 2 tabs. It's working properly (I also modified the label of tabs and a badge on the RelationManager tab) but I can't find a way to set an icon to the "EditRecord" tab (here labelled "Données du chantier"). Here is a screen shot

![image](https://github.com/filamentphp/filament/assets/160853342/f788fa72-6ce1-46a8-95b8-44a320c98e1e)

I've tried to override :
```php
    public function getContentTabIcon(): ?string
    {
        return 'heroicon-s-wrench-screwdriver' ;
    }
```
in the EditRecord class as mentioned in the documentation but this does not work (it is not a parent class method).

Any idea how to add an icon on the first tab ?

Thanks in advance
Didier

**Comment by dissto** [comment]
https://github.com/filamentphp/filament/pull/12214 will probably be merged soon, then it will be supported.

In the meantime you can have a custom view for the header

https://filamentphp.com/docs/3.x/tables/advanced#customizing-the-table-header

**Comment by micraux** [comment]
Works OK for me...

Your parent class is `Job` and the child (relation) class is `Hour`, correct?

By the way, I have 2 questions:
- how do you add the count `3` in your badge tab? I've been searching for more than one hour and can't find a way to do it
- how do you change the parent tab name? Mine is "Modify", I would like to have the model name

**Reply by MagicDidou** [reply]
Hi !

You're right, `Job` is the main class and `Hour` is the "children class" as a `RelationManager` resource with `hasMany / belongsTo` relationship.

* To add the badge at the tab of the Hour, I've just added the following statement in the HoursRelationManager class :
```php
use Illuminate\Database\Eloquent\Model;
 
public static function getBadge(Model $ownerRecord, string $pageClass): ?string
{
   // $ownerModel is of actual type Job
    return $ownerRecord->hours()->count() ;
}
```

* To change the parent tab name you must do it in the EditRecord (or ViewRecord if it's a view) class which represents the page :

```php
public function getContentTabLabel(): ?string
{
    return 'Whatever you want';
}
```

* To change the title of the page (default is Modify) override the getTitle method :
```php

    public function getTitle(): string | Htmlable
    {
        // Could be based on current record data with $this->getRecord()
        return 'Your own title' ;
    }
```

**In return, my question : how did you get the Icon on the parent class ?**

Did

**Comment by micraux** [comment, marked as answer]
Ok, let's share some information, I think we can help each other!

First, thanks for the `getContentTabLabel()` function, it works!

As you are French, you'll understand my Models names ;-) For non French speakers here, my parent Model is `Editeur` (yes, **Editor** in English, that was easy) and the child Models are `Livre` (**Book**) and `Collection` (do I really need to translate?)

Here is a screenshot of my RelationManagers:
![Capture d’écran 2024-04-08 à 14 10 06](https://github.com/filamentphp/filament/assets/46725831/13accb43-28f8-4309-9941-7ba4631851e5)

As you can notice, I managed to add an icon to the children, just adding this line in the Relation Manager:
```php
// in my LivresRelationManager.php file
protected static ?string $icon = 'heroicon-o-book-open';
```

Concerning the icon on the parent class, I just added the `getContentTabIcon()` in the EditEditeur.php file, which doesn't seem to work for you.

Could it be related to your Filament version?
Are you using the v3.0.0 (as you mentioned on this thread) or did you upgrade?

**Reply by MagicDidou** [reply]
Hello (bonjour !),

I'm glad that the `getContentLabel()` worked fine (I had no doubt about it btw).

Regarding the `getContentTabIcon()` I think you are probably right. I've not updated the filament package since I started this new project and the method is the one described in the documentation.

I'll try this this evening and will let you know.

Have a good day and thanks !
(Bonne journée et merci !)

**Reply by MagicDidou** [reply]
I updated my project (to filament V3.2.63) and `getContentTabIcon()` works now !

Thanks

**Comment by himanshumehru** [comment]
Once all the tabs are combined with the content, is there a way to set the default open tab if I didn't want it to be the content tab?

**Reply by MagicDidou** [reply]
To do that, you just have to add a `activeRelationManager` parameter to your Url. Example :

```php
MyResource::getUrl('view', ['record' => $record->id , 'activeRelationManager' => 0]) )
```
This will open the view of the record and will automatically jump to first Tab (=0).

Don't know if this is official...
